tests/unittests/CoreGraphics.drawing/CGContextDrawing_ImageDrawingTests.cpp

```diff
@@ -0,0 +1,243 @@
+    CGRect rect = { { 0, 0 }, { 20, 20 } };
+    _drawTiledImage(GetDrawingContext(), rect, "tiledImageCircleMe.png");
+}
+
+DISABLED_DRAW_TEST_F(CGImageDrawing, TiledImageCustomScaledDownReallyLow, UIKitMimicTest) {
+    CGRect rect = { { 0, 0 }, { 1, 1 } };
+    _drawTiledImage(GetDrawingContext(), rect, "tiledImageCircleMe.png");
+}
+
+DISABLED_DRAW_TEST_F(CGImageDrawing, TiledImageCustomScaled, UIKitMimicTest) {
+    CGRect rect = { { 0, 0 }, { 10, 250 } };
+    _drawTiledImage(GetDrawingContext(), rect, "tiledImageCircleMe.png");
+}
+
+DISABLED_DRAW_TEST_F(CGImageDrawing, TiledImageCustomScaledObscure, UIKitMimicTest) {
+    CGRect rect = { { 0, 0 }, { 253, 13 } };
+    _drawTiledImage(GetDrawingContext(), rect, "tiledImageCircleMe.png");
+}
+
+DISABLED_DRAW_TEST_F(CGImageDrawing, TiledImageCustomScaledAlpha, UIKitMimicTest) {
+    CGRect rect = { { 0, 0 }, { 128, 128 } };
+    CGContextSetAlpha(GetDrawingContext(), 0.88);
+    _drawTiledImage(GetDrawingContext(), rect, "tiledImageCircleMe.png");
+}
+#pragma endregion TiledImageDrawing
+
+#pragma region DrawingToCanvas
+
+#ifdef WINOBJC
+#include "CGContextInternal.h"
+
+DISABLED_DRAW_TEST_F(CGImageDrawing, DrawIntoRect, UIKitMimicTest) {
+    // Draw a portion of an image into a different region.
+    auto drawingConfig = DrawingTestConfig::Get();
+
+    woc::unique_cf<CFStringRef> testFilename{ _CFStringCreateWithStdString(drawingConfig->GetResourcePath("png3.9.png")) };
+    woc::unique_cf<CGImageRef> image{ _CGImageCreateFromPNGFile(testFilename.get()) };
+    ASSERT_NE(image, nullptr);
+
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGAffineTransform flip = CGAffineTransformMakeScale(1, -1);
+    CGAffineTransform shift = CGAffineTransformTranslate(flip, 0, bounds.size.height * -1);
+    CGContextConcatCTM(context, shift);
+
+    _CGContextDrawImageRect(context,
+                            image.get(),
+                            { 0, 0, bounds.size.width / 4, bounds.size.height / 4 },
+                            { 0, 0, bounds.size.width, bounds.size.height });
+}
+#endif
+
+DRAW_TEST_F(CGImageDrawing, DrawAnImage, UIKitMimicTest) {
+    // Load an Image and draw it into the canvas context
+    auto drawingConfig = DrawingTestConfig::Get();
+
+    woc::unique_cf<CFStringRef> testFilename{ _CFStringCreateWithStdString(drawingConfig->GetResourcePath("jpg1.jpg")) };
+    woc::unique_cf<CGImageRef> image{ _CGImageCreateFromJPEGFile(testFilename.get()) };
+    ASSERT_NE(image, nullptr);
+
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGAffineTransform flip = CGAffineTransformMakeScale(1, -1);
+    CGAffineTransform shift = CGAffineTransformTranslate(flip, 0, bounds.size.height * -1);
+    CGContextConcatCTM(context, shift);
+
+    CGContextDrawImage(context, bounds, image.get());
+}
+
+DRAW_TEST_F(CGImageDrawing, RedBox, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
```

Comment:
I did and they show as moved.

---
In reply to: [93534482](https://github.com/Microsoft/WinObjC/pull/1605#discussion_r93534482) [](ancestors = 93534482)